### PR TITLE
Fixed #25417/#24636 -- Added validation for DecimalField digits/Added a field check for invalid default values.

### PR DESCRIPTION
**Pull request by charettes**
Based on a rebased version of #4514

### CODE REVIEW

**Comment by charettes** [comment]
Added a release note and modified @iuliachiriac's patch to avoid creating unnecessary models and abstract the existing form decimal field validation into a validator that can be used at both the model and form layer.

**Comment by charettes** [comment]
I guess failures are related to #5304?

**Comment by charettes** [comment]
Previous review can be found [here](https://github.com/charettes/django/commit/13aacdb6206afc873fe7819f3442a1ace513090d#comments).

File: docs/ref/validators.txt [hunk 1]
```diff
@@ -281,3 +281,19 @@ to, or in lieu of custom ``field.clean()`` methods.
     .. versionchanged:: 1.8
 
        The ``message`` parameter was added.
+
+``DecimalValidator``
+-------------------------
```
Comment (timgraham) [inline docs/ref/validators.txt]: match heading length with underline

File: docs/releases/1.9.txt [hunk 2]
```diff
@@ -511,6 +511,8 @@ Models
 * Added support for referencing annotations in ``QuerySet.distinct()``.
 
 * ``connection.queries`` shows queries with substituted parameters on SQLite.
+* Added a new field check that makes sure :attr:`~django.db.models.Field.default`
```
Comment (timgraham) [inline docs/releases/1.9.txt]: Add newline above.

**Pull request merged**